### PR TITLE
MM-50940 Add community data to looker

### PR DESCRIPTION
**Pull request by sri-byte**
Impact: Creating a marts table for Community data to bring to Looker.

### CODE REVIEW

File: transform/mattermost-analytics/models/marts/web_app/community/community_tracks.sql [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    config({
+        "materialized": "incremental",
+        "cluster_by": ['event_date']
+    })
+}}
+
+WITH community_prod AS (
+    SELECT 
+        {{dbt_utils.star(ref('stg_mm_telemetry_prod__tracks'))}}
+        , received_at::date as received_at_date
```
Comment (ifoukarakis) [inline transform/mattermost-analytics/models/marts/web_app/community/community_tracks.sql:11]: Not sure why `received_at` is cast to a date here. Is it for incremental logic? If yes, this is complicating the query. 

If it's still needed, [incremental strategy](https://docs.getdbt.com/docs/build/incremental-models#configuring-incremental-strategy) also needs to be configured. In this case, the same `received_at_date` might be part of the query, making the merge strategy the most likely approach.

File: transform/mattermost-analytics/models/marts/web_app/community/_community__models.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+version: 2
+
+models:
+  - name: community_tracks
+    description: |
+      Rudderstack default columns from tracks for Community.
+
+    columns:
```
Comment (ifoukarakis) [inline transform/mattermost-analytics/models/marts/web_app/community/_community__models.yml:8]: Please add tests.

File: transform/mattermost-analytics/models/marts/web_app/community/community_tracks.sql [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        , timestamp::date as event_date
+     FROM
+       {{ ref('stg_mm_telemetry_prod__tracks') }} 
+    WHERE server_id = '93mykbogbjfrbbdqphx3zhze5c'
+{% if is_incremental() %}
+    AND received_at_date >= (SELECT MAX(received_at_date) FROM {{ this }})
```
[the remark this answers is not in the document]
Comment (sri-byte) [reply transform/mattermost-analytics/models/marts/web_app/community/community_tracks.sql:17]: Moved source relation into sub query

File: transform/mattermost-analytics/models/marts/web_app/community/community_tracks.sql [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+}}
+
+WITH community_prod AS (
+    SELECT 
+        {{dbt_utils.star(ref('stg_mm_telemetry_prod__tracks'))}}
+        , received_at::date as received_at_date
+        , timestamp::date as event_date
+     FROM
+       {{ ref('stg_mm_telemetry_prod__tracks') }} 
+    WHERE server_id = '93mykbogbjfrbbdqphx3zhze5c'
```
[the remark this answers is not in the document]
Comment (sri-byte) [reply transform/mattermost-analytics/models/marts/web_app/community/community_tracks.sql:15]: Created a new variable community_server_id

File: transform/mattermost-analytics/models/marts/web_app/community/community_tracks.sql (hunk 1 above)
Comment (sri-byte) [reply transform/mattermost-analytics/models/marts/web_app/community/community_tracks.sql:11]: Updated the incremental strategy to append

File: transform/mattermost-analytics/models/marts/web_app/community/_community__models.yml (hunk 2 above)
Comment (sri-byte) [reply transform/mattermost-analytics/models/marts/web_app/community/_community__models.yml:8]: Added tests on event_id column

**Pull request closed without merging**